src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,243 @@
+     */
+    public Storage(String fileName) {
+        this.saveFileName = fileName;
+    }
+
+    /**
+     * Converts LocalDate into "dd/MM/yyyy" string format for saving.
+     *
+     * @param localDate LocalDate to be converted.
+     * @return Formatted String suitable for saving.
+     * @throws DukeException If localDate cannot be parsed into save format.
+     */
+    private static String toSaveDateFormat(LocalDate localDate) throws DukeException {
+        try {
+            return localDate.format(DateTimeFormatter.ofPattern("dd/MM/yyyy"));
+        } catch (DateTimeException dateTimeException) {
+            throw new DukeException(Ui.exceptionInvalidLocalDate());
+        }
+    }
+
+    /**
+     * Converts a Task into a formatted string suitable for saving.
+     *
+     * @param task Task to be converted.
+     * @return Formatted String suitable for saving.
+     * @throws DukeException If date cannot be parsed to save format.
+     */
+    private static String toSaveFormat(Task task) throws DukeException {
+        // Initialize StringBuilder.
+        StringBuilder stringBuilder = new StringBuilder();
+
+        // Get task type, done status of task.
+        String taskType = task.getTaskType();
+        int done = task.getIsDone() ? 1 : 0;
+
+        // Build corresponding save string from task.
+        stringBuilder.append(taskType).append(SAVE_SEPARATOR);
+        stringBuilder.append(done).append(SAVE_SEPARATOR);
+        stringBuilder.append(task.getDescription());
+
+        // duke.task.Deadline tasks have time, so it is obtained and appended via stringBuilder.
+        if (taskType.equals("D")) {
+            Deadline deadline = (Deadline) task;
+            LocalDate localDate = deadline.getTime();
+            String time = toSaveDateFormat(localDate);
+            stringBuilder.append(SAVE_SEPARATOR).append(time);
+        }
+
+        // duke.task.Event tasks have time, so it is obtained and appended via stringBuilder.
+        if (taskType.equals("E")) {
+            Event event = (Event) task;
+            LocalDate localDate = event.getTime();
+            String time = toSaveDateFormat(localDate);
+            stringBuilder.append(SAVE_SEPARATOR).append(time);
+        }
+
+        return stringBuilder.toString();
+    }
+
+    /**
+     * Converts string read from specified file into corresponding tasks based on the string identifier.
+     *
+     * @param save The read string.
+     * @return Task corresponding to the read string.
+     * @throws DukeException if string cannot be parsed to date.
+     * @throws DukeException if string cannot be parsed to integer.
+     * @throws DukeException if string does not have sufficient parts for the task it represents.
+     */
+    private static Task parseSaveFormat(String save) throws DukeException {
+        // Split save string by the save separator.
+        String[] saveSplit = save.split(SAVE_SEPARATOR);
```

Comment:
Perhaps this variable name could be made into a noun too?

Reply:
Thanks for the suggestion! I agree that it would be more appropriate to use a noun as the name of this variable and I will make the changes in the next push.